lib/laforge/activerecord.rb

```diff
@@ -86,41 +90,13 @@ def update_attribute_source_priority(attribute, source, priority)
 
     # Returns a list of the entries matching the filters
     def filter_loaded_data_entries(attributes: nil, sources: nil, present: nil)
-      list = data_entries.to_a
-      return list if list.blank?
-
-      unless attributes.nil?
-        attributes = Array.wrap(attributes).map(&:to_s)
-        list.select! {|data_entry| attributes.include?(data_entry.attribute_name) }
-      end
-
-      unless sources.nil?
-        source_ids = Array.wrap(sources).map {|source| source_id_from(source) }.compact
-        list.select! {|data_entry| source_ids.include?(data_entry.source_id) }
-      end
-
-      list.select!(&:present?) if present == true
-      list.reject!(&:present?) if present == false
+      list = data_entries
```

Comment:
I think we should make sure we return a new array even if no filtering takes place or we'll sometimes have a relation and sometimes have an array, which may have shared method names but one actually updates the database where the other just changes the contents of the array.